rbc/external.py

```diff
@@ -79,7 +77,16 @@ def name(self):
 
     def match_signature(self, atypes):
         # Code here is the same found in remotejit.py::Signature::best_match
-        device = "CPU" if TargetInfo().is_cpu else "GPU"
+        target_info = TargetInfo()
+        device = "CPU" if target_info.is_cpu else "GPU"
+        if device not in self._signatures:
+            satypes = ", ".join(map(str, atypes))
+            compile_target = target_info.get_compile_target()
+            raise TypeError(
+                f"{compile_target=}: no signatures with argument types `{satypes}` and"
+                f" device `{device}` while processing `{self.name}`"
+                " (perhaps the compile target cannot support the device?)"
+            )
```

Comment:
Isn't the `TypeError` below doing the same thing? Raise an exception if no function for the given signature and device is found.

Reply:
Here, the type error is raised because a signature with the target device is missing while below the error is raised because a signature with the given argument types is missing.

The first case happens when, say, when target info is GPU while a signature has `TableFunctionManager` argument that is available only for the CPU target (hence, there is no `GPU` device key in `self._signatures`).

Reply:
I have revised this code. So, please review the update code.